demo/clusters/kind/scripts/common.sh

```diff
@@ -46,7 +47,11 @@ DRIVER_IMAGE_VERSION=$(from_versions_mk "VERSION")
 : ${KIND_CLUSTER_CONFIG_PATH:="${SCRIPTS_DIR}/kind-cluster-config.yaml"}
 
 # The derived name of the kind image to build
-: ${KIND_IMAGE_BASE_TAG:="v20230515-01914134-containerd_v1.7.1"}
-: ${KIND_IMAGE_BASE:="gcr.io/k8s-staging-kind/base:${KIND_IMAGE_BASE_TAG}"}
-: ${KIND_IMAGE:="kindest/node:${KIND_K8S_TAG}-${KIND_IMAGE_BASE_TAG}"}
+${NVIDIA_DRIVER_ROOT:+--set nvidiaDriverRoot=${NVIDIA_DRIVER_ROOT}}
+
+## KIND_IMAGE_BASE_TAG can be specified to build specific requirements into a kind image.
+## This should in general not be required.
+# : ${KIND_IMAGE_BASE_TAG:="v20230515-01914134-containerd_v1.7.1"}
+: ${KIND_IMAGE_BASE:="${KIND_IMAGE_BASE_TAG:+gcr.io/k8s-staging-kind/base:${KIND_IMAGE_BASE_TAG}}"}
+: ${KIND_IMAGE:="kindest/node:${KIND_K8S_TAG}${KIND_IMAGE_BASE_TAG:+-${KIND_IMAGE_BASE_TAG}}"}
```

Comment:
This section is confusing to me. Why do we have `KIND_IMAGE_BASE_TAG` but it is commented out. We then clearly set a `KIND_IMAGE_BASE` (though I guess it equals "" since `KIND_IMAGE_BASE_TAG ` is commented out?). Then in `KIND_IMAGE` we only conditionally include `-${KIND_IMAGE_BASE_TAG}` if `KIND_IMAGE_BASE_TAG` is set (which according to the logic above it never is).

The only thing I can figure is that this is designed to support someone setting `KIND_IMAGE_BASE_TAG` on the command line, which I don't think we want to do.

Can we simplify this to remove all of this inline cnditional logic (maybe leaving a separate comment block to show what *would* be necessary if we ever need to support a `KIND_IMAGE_BASE_TAG` in the future again).

Reply:
OK. I have cleaned this up a bit. We only specify the full kind image here and then the note on how to use a custom base image is included where we build the image.

I think it's less likely that we will need custom base images since we don't specifically need runtime features for this to work.